paddle/go/cclient/cclient.go

```diff
@@ -83,29 +84,46 @@ func cArrayToSlice(p unsafe.Pointer, len int) []byte {
 	return (*[1 << 30]byte)(p)[:len:len]
 }
 
+type selector bool
+
+func (s selector) Select() bool {
+	return bool(s)
+}
+
+type lister []pserver.Server
+
+func (l lister) List() []pserver.Server {
+	return l
```

Comment:
maybe we should implement this interface bases on ectd? for example, list pserver names from etcd value for fault tolerant pserver?